core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -165,50 +165,36 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
-    } yield RawlsBillingProjectResponse(
+  private def makeBillingProjectResponse(projectRoles: Set[ProjectRole], billingProject: RawlsBillingProject) =
+    RawlsBillingProjectResponse(
       billingProject.projectName,
+      billingProject.status,
       billingProject.billingAccount,
       billingProject.servicePerimeter,
       billingProject.invalidBillingAccount,
-      projectRoles,
-      workspacesWithCorrectBillingAccount.map(_.toWorkspaceName).toSet,
-      workspacesWithIncorrectBillingAccount.map(workspace => WorkspaceBillingAccount(workspace.toWorkspaceName, workspace.currentBillingAccountOnGoogleProject)).toSet)
-  }
-
-  // This returns two seqs of Workspace. The first seq is the ones with the correct billing account, the second seq is the ones with incorrect billing accounts
-  private def loadAndPartitionWorkspacesByMatchingBillingProjectBillingAccount(billingProject: RawlsBillingProject): Future[(Seq[Workspace], Seq[Workspace])] = {
-    dataSource.inTransaction { dataAccess =>
-      dataAccess.workspaceQuery.listWithBillingProject(billingProject.projectName).map(workspaces => {
-        workspaces.partition(workspace => workspace.currentBillingAccountOnGoogleProject == billingProject.billingAccount)
-      })
-    }
-  }
+      projectRoles
+    )
 
-  def listBillingProjectsV2(): Future[PerRequestMessage] = {
+  def listBillingProjectsV2(): Future[PerRequestMessage] =
     for {
       samUserResources <- samDAO.listUserResources(SamResourceTypeNames.billingProject, userInfo)
-      projectsInDB <- dataSource.inTransaction { dataAccess => dataAccess.rawlsBillingProjectQuery.getBillingProjects(samUserResources.map(resource => RawlsBillingProjectName(resource.resourceId)).toSet) }
-    } yield {
-      val projectResponses = constructBillingProjectResponses(samUserResources, projectsInDB)
-      RequestComplete(projectResponses)
-    }
-  }
+      projectNames = samUserResources.map(r => RawlsBillingProjectName(r.resourceId)).toSet
+      projectsInDB <- dataSource.inTransaction { dataAccess =>
+        dataAccess.rawlsBillingProjectQuery.getBillingProjects(projectNames)
+      }
+    } yield RequestComplete(constructBillingProjectResponses(samUserResources, projectsInDB))
 
-  private def constructBillingProjectResponses(samUserResources: Seq[SamUserResource], billingProjectsInRawlsDB: Seq[RawlsBillingProject]): Future[List[RawlsBillingProjectResponse]] = {
+  private def constructBillingProjectResponses(samUserResources: Seq[SamUserResource], billingProjectsInRawlsDB: Seq[RawlsBillingProject]): List[RawlsBillingProjectResponse] = {
     val projectsByName = billingProjectsInRawlsDB.map(p => p.projectName -> p).toMap
-    for {
-      billingProjectResponses <- samUserResources.toList.traverse { samUserResource =>
-        val allSamRoles = samUserResource.direct.roles ++ samUserResource.inherited.roles
```

Comment:
could we add `allSamRoles ` back? Without it, I wouldn't have realized that this covers all sam roles

Reply:
> jenkins retest -- test failure is the flaky sam 500 error (https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-fiab-test-runner/37044/testReport/junit/org.broadinstitute.dsde.test.api/RawlsApiSpec/Rawls_should_be_able_to_create_workspace_and_run_sub_workflow_tasks_in_non_US_regions/)

Thank you so much!

Reply:
I don't think it adds value - there's no way to enforce that this is indeed all sam roles.